### ISSUE TITLE
Error: 'The option `options` must be used together with device class `enum`, got `device_class` 'None'

### ISSUE THREAD

**Issue by timbba80** [syssi/homeassistant-goecharger-mqtt#180]
I've seen these error messages in the home assistant logs for a while now. Everything works fine though. Am I the only one?

Home Assistant 2024.11.3

Logger: homeassistant.components.mqtt.entity
Source: components/mqtt/entity.py:155
integration: MQTT ([documentation](https://www.home-assistant.io/integrations/mqtt), [issues](https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+mqtt%22))
First occurred: 19:54:01 (2 occurrences)
Last logged: 19:54:01

Error 'The option `options` must be used together with device class `enum`, got `device_class` 'None'' when processing MQTT discovery message topic: 'homeassistant/sensor/go-e_xxxxxx/car_state/config', message: '{'value_template': "{{ ['Unknown','Idle','Charging','WaitCar','Complete','Error'][value_json|int ]}}", 'state_topic': 'go-eCharger/xxxxx/car', 'options': ['Unknown', 'Idle', 'Charging', 'WaitCar', 'Complete', 'Error'], 'command_template': "{% set o = {'Unknown': 0,'Idle': 1,'Charging': 2,'WaitCar': 3,'Complete': 4,'Error': 5} %}{{ o[value] if value in o else value }}", 'unique_id': 'go-e_xxxxx_car_state', 'name': 'Car State', 'device': {'model': 'go-eCharger_V4', 'identifiers': ['go-e_xxxxx'], 'sw_version': '56.8', 'name': 'go-eCharger_xxxxxx', 'manufacturer': 'go-e'}}'

Error 'The option `options` must be used together with device class `enum`, got `device_class` 'None'' when processing MQTT discovery message topic: 'homeassistant/sensor/go-e_xxxxxx/error_state/config', message: '{'value_template': "{{['None','FiAc','FiDc','Phase','Overvolt','Overamp','Diode','PpInvalid','GndInvalid','ContactorStuck','ContactorMiss','StatusLockStuckOpen','StatusLockStuckLocked','FiUnknown','Unknown','Overtemp','NoComm','CpInvalid'][value_json|int ]}}", 'state_topic': 'go-eCharger/xxxxxx/err', 'options': ['None', 'FiAc', 'FiDc', 'Phase', 'Overvolt', 'Overamp', 'Diode', 'PpInvalid', 'GndInvalid', 'ContactorStuck', 'ContactorMiss', 'StatusLockStuckOpen', 'StatusLockStuckLocked', 'FiUnknown', 'Unknown', 'Overtemp', 'NoComm', 'CpInvalid'], 'command_template': "{% set o = {'None': 0,'FiAc': 1,'FiDc': 2,'Phase': 3,'Overvolt': 4,'Overamp': 5,'Diode': 6,'PpInvalid': 7,'GndInvalid': 8,'ContactorStuck': 9,'ContactorMiss': 10,'StatusLockStuckOpen': 11,'StatusLockStuckLocked': 12,'FiUnknown': 13,'Unknown': 14,'Overtemp': 15,'NoComm': 16,'CpInvalid': 17} %}{{ o[value] if value in o else value }}", 'unique_id': 'go-e_xxxxxx_error_state', 'name': 'Error State', 'device': {'model': 'go-eCharger_V4', 'identifiers': ['go-e_xxxxxx'], 'sw_version': '56.8', 'name': 'go-eCharger_xxxxxx, 'manufacturer': 'go-e'}}'

**Comment by AndreasLehmann** [comment]
same here

**Comment by syssi** [comment]
It looks like you aren't using this integration/project. The error is triggered by the `mqtt` integration because the auto-discovered MQTT topics of the Go-e charger aren't well defined.

**Comment by AndreasLehmann** [comment]
You are right. The go-e charger is listed in the MQTT Integration and in the goecharger-Integration. Can I disable the MQTT Integration without affecting the  goecharger-Integration ?

**Comment by syssi** [comment]
The MQTT integration is an important dependency/requirement. You can disable or remove the auto-discovered MQTT device instead.
